python/src/acp_sdk/server/app.py

```diff
@@ -91,6 +91,11 @@ async def read_agent(name: AgentName) -> AgentReadResponse:
         agent = find_agent(name)
         return AgentModel(name=agent.name, description=agent.description, metadata=agent.metadata)
 
+    @app.get("/agents/{name}/healthcheck")
+    async def ping_agent(name: AgentName) -> str:
+        find_agent(name)
+        return "OK"
```

Comment:
Agent specific healthcheck would need proper implementation in the agent class. In this form, it is a provider-wide healthcheck that can't be used as such without the knowledge of the agent name.

Just go with `/healthcheck` endpoint, that can be easily leveraged by kubernetes.